ios/Classes/PdftronFlutterPlugin.m

```diff
@@ -126,81 +131,133 @@ + (void)configureDocumentViewController:(PTDocumentViewController*)documentViewC
         return;
     }
     
-    //convert from json to dict
-    NSData* jsonData = [config dataUsingEncoding:NSUTF8StringEncoding];
-    id foundationObject = [NSJSONSerialization JSONObjectWithData:jsonData options:kNilOptions error:Nil];
+    PTFlutterViewController* flutterViewController = (PTFlutterViewController*)documentViewController;
     
-    NSAssert([foundationObject isKindOfClass:[NSDictionary class]], @"config JSON object not in expected dictionary format.");
+    [flutterViewController initViewerSettings];
     
-    if([foundationObject isKindOfClass:[NSDictionary class]])
-    {
-        //convert from json to dict
-        NSData* jsonData = [config dataUsingEncoding:NSUTF8StringEncoding];
-        id foundationObject = [NSJSONSerialization JSONObjectWithData:jsonData options:kNilOptions error:Nil];
-
-        NSAssert([foundationObject isKindOfClass:[NSDictionary class]], @"config JSON object not in expected dictionary format.");
+    //convert from json to dict
+    id foundationObject = [PdftronFlutterPlugin PT_JSONStringToId:config];
+    
+    if (![foundationObject isKindOfClass:[NSNull class]]) {
+        
+        NSDictionary* configPairs = [PdftronFlutterPlugin PT_idAsNSDict:foundationObject];
         
-        if([foundationObject isKindOfClass:[NSDictionary class]])
+        if(configPairs)
         {
-            NSDictionary* configPairs = (NSDictionary*)foundationObject;
+            
+            NSError* error;
             
             for (NSString* key in configPairs.allKeys) {
                 if([key isEqualToString:PTDisabledToolsKey])
                 {
-                    id toolsToDisable = configPairs[PTDisabledToolsKey];
-                    if(![toolsToDisable isEqual:[NSNull null]])
-                    {
-                        NSAssert([toolsToDisable isKindOfClass:[NSArray class]], @"disabledTools JSON object not in expected array format.");
-                        if([toolsToDisable isKindOfClass:[NSArray class]])
-                        {
-                            [self disableTools:(NSArray*)toolsToDisable documentViewController:documentViewController];
-                        }
-                        else
-                        {
-                            NSLog(@"disabledTools JSON object not in expected array format.");
-                        }
+                    
+                    NSArray* toolsToDisable = [PdftronFlutterPlugin getConfigValue:configPairs configKey:PTDisabledToolsKey class:[NSArray class] error:&error];
+                    
+                    if (!error && toolsToDisable) {
```

Comment:
We should at least have a log statement if there is an error. Preferably it is reported programmatically, but if there isn't at least a log statement, it could be potentially quite time consuming to track down an error.

Same applies for following error objects.

Reply:
Here: https://github.com/PDFTron/pdftron-flutter/blob/1f0a8b0ea66145bb491bc9dcc9359e11c1acbe08/ios/Classes/PdftronFlutterPlugin.m#L233-L235